src/control/client/network.go

```diff
@@ -0,0 +1,173 @@
+//
+// (C) Copyright 2018-2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+
+	ctlpb "github.com/daos-stack/daos/src/control/common/proto/ctl"
+)
+
+const (
+	networkScanTimeout = 10 * time.Second
+)
+
+type ProviderListResult struct {
+	providerList string
+}
+
+type NetworkScanResult struct {
+	provider string
+	device   string
+	numanode uint
+	err      error
+}
+
+type NetworkScanResultMap map[string][]NetworkScanResult
+
+func (nsrm NetworkScanResultMap) String() string {
+	var buf bytes.Buffer
+	servers := make([]string, 0, len(nsrm))
+
+	for server := range nsrm {
+		servers = append(servers, server)
+	}
+	sort.Strings(servers)
+
+	for _, server := range servers {
+		fmt.Fprintf(&buf, "%s:\n%s\n", server, nsrm[server])
+	}
+
+	return buf.String()
+}
+
+func (nsr NetworkScanResult) String() string {
+	var buf bytes.Buffer
+	if nsr.err != nil {
+		fmt.Fprintf(&buf, "\n\tError: %v\n", nsr.err)
+		return buf.String()
+	}
+	fmt.Fprintf(&buf, "\n\tfabric_iface: %s\n\tprovider: %s\n\tpinned_numa_node: %d\n", nsr.device, nsr.provider, nsr.numanode)
```

Comment:
are you sure you want new-lines between the parameter values? maybe just separate by space to have one result per line

Reply:
I used the same multi-line formatting on the daos_server side, and implemented the same formatting here.  I imagine that we might want to tweak things when we have many servers reporting data, but I don't know exactly what that should look like yet.  If we put the output on a single line, it might be difficult to look at when there are more than a few items to report per system scan.  With the current scheme, the formatting looks good even if there are 10's of devices to report.  But, as you are getting at, it might be unwieldy when there are lots of servers.  I think it's a good discussion to have as we work on the topic to 'scale/format the output for ~100 servers'.

Reply:
Oh, I misread what you're asking for.  Yes, it would be OK to put the results for a single record on a single line.  Still, I'm wondering if we want to make that change right now or wait until we specifically address formatting the output of our tools for ~100 servers.

Reply:
If you feel strongly about putting it on a single line right now, I will make the edit and resubmit.